presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcConnector.java

```diff
@@ -80,7 +84,8 @@ public JdbcConnector(
             FunctionMetadataManager functionManager,
             StandardFunctionResolution functionResolution,
             RowExpressionService rowExpressionService,
-            JdbcClient jdbcClient)
+            JdbcClient jdbcClient,
+            Optional<JdbcSessionPropertiesProvider> sessionPropertiesProvider)
```

Comment:
How about having this parameter in the form of Set< JdbcSessionPropertiesProvider> ?

IMO, that allows us to have multiple session provider implementations as per the use cases